src/controllers/auth.controller.js

```diff
@@ -0,0 +1,60 @@
+const User = require("../models/User");
+const {
+  generateAccessToken,
+  generateRefreshToken,
+} = require("../utils/jwtUtils");
+
+const { ONE_HOUR_IN_MILLISECONDS } = require("../constants/jwtConstants");
+
+exports.signIn = async function (req, res, next) {
+  const { email, photoURL, displayName } = req.body;
+
+  try {
+    let user = await User.findOne({ email });
+
+    if (!user) {
+      user = new User({
+        email,
+        photoURL,
+        displayName,
+        isAdmin: false,
+      });
+    }
+
+    const accessToken = generateAccessToken(user);
+    const refreshToken = generateRefreshToken(user);
+
+    user.refreshToken = refreshToken;
+
+    await user.save();
+
+    res.status(201).cookie("accessToken", accessToken, {
+      maxAge: ONE_HOUR_IN_MILLISECONDS,
+      httpOnly: true,
+    });
+
+    res.send({ result: "ok", message: "login successful!", user });
+  } catch (error) {
+    console.log(error);
+  }
+};
+
+exports.signOut = async function (req, res, next) {
+  res.clearCookie("accessToken");
+  res.clearCookie("refreshToken");
+
+  res.send({ result: "ok", message: "logout successful" });
+};
+
+exports.check = async function (req, res, next) {
+  console.log("check!", req.user);
+
+  if (!req.user) {
+    res.status(200).json({ result: false });
+    return;
+  }
+
+  const user = await User.findById(req.user);
```

Comment:
lean() 사용이 가능해 보입니다!
```suggestion
  const user = await User.findById(req.user).lean();
```